Sources/StreamChat/Controllers/ChannelController/LivestreamChannelController.swift

```diff
@@ -0,0 +1,502 @@
+        updateChannelData(channelQuery: query, completion: completion)
+    }
+    
+    /// Loads next messages from backend.
+    /// - Parameters:
+    ///   - messageId: ID of the current first message. You will get messages `newer` than the provided ID.
+    ///   - limit: Limit for page size. By default it is 25.
+    ///   - completion: Called when the network request is finished.
+    public func loadNextMessages(
+        after messageId: MessageId? = nil,
+        limit: Int? = nil,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        guard cid != nil else {
+            completion?(ClientError.ChannelNotCreatedYet())
+            return
+        }
+        
+        let messageId = messageId ?? paginationStateHandler.state.newestFetchedMessage?.id ?? messages.first?.id
+        guard let messageId = messageId else {
+            completion?(ClientError.ChannelEmptyMessages())
+            return
+        }
+        
+        guard !hasLoadedAllNextMessages && !isLoadingNextMessages else {
+            completion?(nil)
+            return
+        }
+        
+        let limit = limit ?? channelQuery.pagination?.pageSize ?? .messagesPageSize
+        var query = channelQuery
+        query.pagination = MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
+        
+        updateChannelData(channelQuery: query, completion: completion)
+    }
+    
+    /// Load messages around the given message id.
+    /// - Parameters:
+    ///   - messageId: The message id of the message to jump to.
+    ///   - limit: The number of messages to load in total, including the message to jump to.
+    ///   - completion: Callback when the API call is completed.
+    public func loadPageAroundMessageId(
+        _ messageId: MessageId,
+        limit: Int? = nil,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        guard !isLoadingMiddleMessages else {
+            completion?(nil)
+            return
+        }
+        
+        let limit = limit ?? channelQuery.pagination?.pageSize ?? .messagesPageSize
+        var query = channelQuery
+        query.pagination = MessagesPagination(pageSize: limit, parameter: .around(messageId))
+        
+        updateChannelData(channelQuery: query, completion: completion)
+    }
+    
+    /// Cleans the current state and loads the first page again.
+    /// - Parameter completion: Callback when the API call is completed.
+    public func loadFirstPage(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        var query = channelQuery
+        query.pagination = .init(
+            pageSize: channelQuery.pagination?.pageSize ?? .messagesPageSize,
+            parameter: nil
+        )
+        
+        // Clear current messages when loading first page
+        messages = []
+        
+        updateChannelData(channelQuery: query, completion: completion)
+    }
+
+    /// Marks the channel as read.
+    public func markRead(completion: ((Error?) -> Void)? = nil) {
```

Comment:
Not sure if we should support this - maybe better to just log a warning that marking read is not supported?

Reply:
Does livestream channels don't have a read count? Otherwise I can remove this

Reply:
Done